src/re.rs

```diff
@@ -655,20 +656,32 @@ impl Regex {
     }
 }
 
-pub enum NamesIter<'a> {
-    Native(::std::slice::Iter<'a, Option<&'static str>>),
-    Dynamic(::std::slice::Iter<'a, Option<String>>)
+/// Yields the names of all possible captures.
+/// `None` indicates an unnamed capture; the first element
+/// (capture 0, the whole matched region) is always unnamed.
+///
+/// `'r` is the lifetime of the compiled expression.
+pub enum CaptureNames<'r> {
+    Native(::std::slice::Iter<'r, Option<&'static str>>),
+    Dynamic(::std::slice::Iter<'r, Option<String>>)
 }
 
-impl<'a> Iterator for NamesIter<'a> {
-    type Item=Option<String>;
+impl<'r> Iterator for CaptureNames<'r> {
+    type Item=Option<&'r str>;
+
+    fn next(&mut self) -> Option<Option<&'r str>> {
+        match *self {
+            CaptureNames::Native(ref mut i) =>
+                i.next().map(|o| *o),
+            CaptureNames::Dynamic(ref mut i) =>
+                i.next().as_ref().map(|o| o.as_ref().map(|s| s.as_ref())),
```

Comment:
Ooops, I missed this before I merged. It's fine how it is I guess. I don't think I have a strong preference either way.